arwen/contexts/runtime.go

```diff
@@ -83,10 +80,27 @@ func (context *runtimeContext) StartWasmerInstance(contract []byte, gasLimit uin
 	}
 
 	context.instance = newInstance
+
+	idContext := arwen.AddHostContext(context.host)
+	context.instance.SetContextData(idContext)
+
+	if context.verifyCode {
+		context.verifyCode = false
+		err = context.VerifyContractCode()
+		if err != nil {
+			context.instance = nil
```

Comment:
I should have written `context.CleanInstance()`, so this is a mistake indeed, but `CleanInstance()` will set the instance to `nil` anyway, because it has to. It's just like any failed instantiation - the desired object has not been constructed, it is `nil`, and an error was returned. There's nothing to do afterwards.

`StartWasmerInstance()` must either start an instance completely, or fail completely, which includes failing due to invalid contract code as well. The returned error must be handled anywhere `StartWasmerInstance()` is called (as is normal). An error means that you can't use the instance at all, regardless of why it couldn't start.